addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4763,47 +4762,76 @@ registry['sizing_x'] = registry.sizing.extend({
      * @override
      */
     _getSize: function () {
+        const isMobileView = weUtils.isMobile(this);
+        const resolutionModifier = isMobileView ? '' : 'lg-';
         var width = this.$target.closest('.row').width();
         var gridE = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
         var gridW = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11];
         this.grid = {
-            e: [_.map(gridE, v => ('col-lg-' + v)), _.map(gridE, v => width / 12 * v), 'width'],
-            w: [_.map(gridW, v => ('offset-lg-' + v)), _.map(gridW, v => width / 12 * v), 'margin-left'],
+            e: [
+                gridE.map(v => (`col-${resolutionModifier}${v}`)),
+                gridE.map(v => width / 12 * v),
+                'width',
+            ],
+            w: [
+                gridW.map(v => (`offset-${resolutionModifier}${v}`)),
+                gridW.map(v => width / 12 * v),
+                'margin-left',
+            ],
         };
         return this.grid;
     },
     /**
      * @override
      */
     _onResize: function (compass, beginClass, current) {
+        const isMobileView = weUtils.isMobile(this);
+        const resolutionModifier = isMobileView ? '' : 'lg-';
+        const resolutionModifierWithXL = isMobileView ? '' : '(?:lg-|xl-)';
         if (compass === 'w' || compass === 'e') {
-            const beginOffset = Number(beginClass.match(/offset-lg-([0-9-]+)|$/)[1] || beginClass.match(/offset-xl-([0-9-]+)|$/)[1] || 0);
+            const offsetRegexWithXL = new RegExp(`\\boffset-${resolutionModifierWithXL}([0-9-]+)|$`);
+            const offsetRegex = new RegExp(`\\boffset-${resolutionModifier}([0-9-]+)|$`);
+            const colRegex = new RegExp(`\\bcol-${resolutionModifier}(\\d{1,2})|$`);
+
+            const beginOffset = Number(beginClass.match(offsetRegexWithXL)[1] || 0);
+
+            const targetEl = this.$target[0];
 
             if (compass === 'w') {
                 // don't change the right border position when we change the offset (replace col size)
-                var beginCol = Number(beginClass.match(/col-lg-([0-9]+)|$/)[1] || 0);
-                var offset = Number(this.grid.w[0][current].match(/offset-lg-([0-9-]+)|$/)[1] || 0);
+                const beginCol = Number(beginClass.match(colRegex)[1] || 0);
+                let offset = Number(this.grid.w[0][current].match(offsetRegex)[1] || 0);
                 if (offset < 0) {
                     offset = 0;
                 }
-                var colSize = beginCol - (offset - beginOffset);
+                let colSize = beginCol - (offset - beginOffset);
                 if (colSize <= 0) {
                     colSize = 1;
                     offset = beginOffset + beginCol - 1;
                 }
-                this.$target.attr('class', this.$target.attr('class').replace(/\s*(offset-xl-|offset-lg-|col-lg-)([0-9-]+)/g, ''));
```

Comment:
The regex seemed already wrong actually, not worth a fix in stable I guess but time to fix it :)